### PR TITLE
fix(web-components): set heading levels on Hero component

### PR DESCRIPTION
**Pull request by GCHQ-Developer-299**
Set heading prop to h2 and secondaryHeading to h3 semantic elements in Hero component

BREAKING CHANGE: Heading hierarchy of the component has changed, which might affect accessibility compliance when integrated

## Related issue
re #1687 

### General 

- [x] Changes to docs package checked and committed.
- [x] All acceptance criteria reviewed and met. 

### Testing

- [x] Visual testing against Figma component specification completed. 

### Accessibility 

- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### Testing content extremes

- [x] All prop combinations work without issue. 
- [x] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.
- [x] Controlled and uncontrolled input components tested.

### CODE REVIEW

File: packages/react/cypress-image-diff-screenshots/baseline/IcRadio.cy.tsx-conditional-dynamic-high-contrast.png
Comment (gd2910) [inline packages/react/cypress-image-diff-screenshots/baseline/IcRadio.cy.tsx-conditional-dynamic-high-contrast.png:1]: The changed image is incorrect as the white ring around the option appears at the point of selection. Leads me to believe that the CI test is taking the image too soon.

Comment (GCHQ-Developer-299) [reply packages/react/cypress-image-diff-screenshots/baseline/IcRadio.cy.tsx-conditional-dynamic-high-contrast.png:1]: The test does click the second radio button's container before taking the snapshot, which seems right to me - or am i misunderstanding the test?
```js
it("should render conditional dynamic IcRadio in high contrast mode", () => {
    mount(<ConditionalDynamic />);

    cy.checkHydrated(RADIO_GROUP_SELECTOR);
    cy.get(RADIO_SELECTOR).eq(1).find(".container").click();

    cy.compareSnapshot({
      name: "conditional-dynamic-high-contrast",
      testThreshold: setThresholdBasedOnEnv(DEFAULT_TEST_THRESHOLD + 0.064),
      delay: 1000,
    });
```

Comment (GCHQ-Developer-299) [reply packages/react/cypress-image-diff-screenshots/baseline/IcRadio.cy.tsx-conditional-dynamic-high-contrast.png:1]: We've talked about it and the issue is that white outer circle around the Option 2 radio button (which appears for an instant when clicked) - it seems the delay parameter is not doing anything to delay the screenshot. 

Suggested fix is to remove the delay and add a wait command before the screenshot.

Comment (GCHQ-Developer-299) [reply packages/react/cypress-image-diff-screenshots/baseline/IcRadio.cy.tsx-conditional-dynamic-high-contrast.png:1]: In the end it seems that this test is behaving correctly (the white ring is meant to be there when a radio button is clicked and focussed)

File: packages/web-components/src/components/ic-hero/test/basic/__snapshots__/ic-hero.spec.ts.snap [hunk 1]
```diff
@@ -1,14 +1,16 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`ic-hero component should render small: renders-small-variant 1`] = `
-<ic-hero class="ic-hero-small" heading="Test title" small="" subheading="Test text">
```
Comment (GCHQ-Developer-530) [inline packages/web-components/src/components/ic-hero/test/basic/__snapshots__/ic-hero.spec.ts.snap:4]: I think this test needs updating in the spec.ts file to have size="small" now that the small prop has been removed

Comment (GCHQ-Developer-299) [reply packages/web-components/src/components/ic-hero/test/basic/__snapshots__/ic-hero.spec.ts.snap:4]: Good spot! I've updated the test :)

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Comment by GCHQ-Developer-847** [comment]
Should the removal of the `small` prop be mentioned as a breaking change as well in the commit message?

**Comment by GCHQ-Developer-299** [comment]
> Should the removal of the `small` prop be mentioned as a breaking change as well in the commit message?

Good thinking! Have updated the web components commit message :)

**Review by GCHQ-Developer-847 — APPROVED** [review]

**Pull request merged**